internal/armada/repository/job.go

```diff
@@ -526,6 +541,33 @@ func (repo *RedisJobRepository) ExpireLeases(queue string, deadline time.Time) (
 	return expired, nil
 }
 
+func (repo *RedisJobRepository) AddRetryAttempt(jobId string) error {
+	exists, err := repo.db.Exists(jobRetriesPrefix + jobId).Result()
+	if err != nil {
+		return err
+	}
+	if exists == 0 {
+		return fmt.Errorf("RedisJobRepository.AddRetryAttempt: job with id %q does not exist", jobId)
+	}
+
+	_, err = repo.db.Incr(jobRetriesPrefix + jobId).Result()
```

Comment:
Increment creates the key if it does not exist, so you don't need the logic above.